### PR TITLE
SPCTR-2757:Shift useDeviceType function in higher order component

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description

Shifting useDeviceType to higher order component.

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/slider/edit.js [hunk 1]
```diff
@@ -111,4 +111,4 @@ const uagbSlideClass = createHigherOrderComponent( ( BlockListBlock ) => {
 	};
 }, 'uagbSlideClass' );
 
-addFilter( 'editor.BlockListBlock', 'uagb/slider-child', uagbSlideClass );
+addFilter( 'editor.BlockListBlock', 'uagb/slider-child', uagbSlideClass );
```
Comment (rahulvarma722) [inline src/blocks/slider/edit.js:114]: Remove unwanted changes from here

File: src/blocks/social-share-child/edit.js [hunk 2]
```diff
@@ -7,16 +7,14 @@
...
 import { select } from '@wordpress/data';
 import Settings from './settings';
 import Render from './render';
 import DynamicCSSLoader from '@Components/dynamic-css-loader';
 import { compose } from '@wordpress/compose';
 import AddStaticStyles from '@Controls/AddStaticStyles';
 const SocialShareChildComponent = ( props ) => {
-	const deviceType = useDeviceType();
-	const { isSelected, setAttributes, attributes, clientId, name } = props;
+	const { isSelected, setAttributes, attributes, clientId, name, deviceType, } = props;
```
Comment (rahulvarma722) [inline src/blocks/social-share-child/edit.js:17]: Add proper indentation and remove comma after deviceType

File: src/blocks/timeline/content-timeline-child/edit.js [hunk 3]
```diff
@@ -3,13 +3,11 @@
...
 
 import { useEffect } from '@wordpress/element';
-import { useDeviceType } from '@Controls/getPreviewType';
 import Settings from './settings';
 import Render from './render';
 
 const ContentTimelineChildComponent = ( props ) => {
-	const deviceType = useDeviceType();
-	const { isSelected, setAttributes } = props;
+	const { isSelected, setAttributes, deviceType, } = props;
```
Comment (rahulvarma722) [inline src/blocks/timeline/content-timeline-child/edit.js:10]: remove comma after deviceType and maintain indentation

File: src/blocks/slider/edit.js (hunk 1 above)
Comment (rajmohanreddy774) [reply src/blocks/slider/edit.js]: Removed

**Review by rahulvarma722 — APPROVED** [review]

File: src/blocks/timeline/content-timeline-child/edit.js (hunk 3 above)
Comment (rajmohanreddy774) [reply src/blocks/timeline/content-timeline-child/edit.js]: Removed

File: src/blocks/social-share-child/edit.js (hunk 2 above)
Comment (rajmohanreddy774) [reply src/blocks/social-share-child/edit.js]: Removed

File: src/blocks/slider/edit.js
Comment (RajkiranBagal) [inline src/blocks/slider/edit.js:1]: Should we remove`import { useDeviceType } from '@Controls/getPreviewType';`  from this file?

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/slider/edit.js
Comment (rajmohanreddy774) [reply src/blocks/slider/edit.js:1]: No we cant remove this intentionally left it for some custom functionality.

**Pull request merged**